src/File/Path/DefaultProcessor.php

```diff
@@ -57,7 +58,7 @@ class DefaultProcessor implements ProcessorInterface
      * @param string           $field the field for which data will be saved
      * @param array            $settings the settings for the current field
      */
-    public function __construct(Table $table, EntityInterface $entity, $data, string $field, array $settings)
+    public function __construct(Table $table, EntityInterface $entity, UploadedFile $data, string $field, array $settings)
```

Comment:
You should typehint against the interface `Psr\Http\Message\UploadedFileInterface` not the concrete class.